segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -454,7 +455,21 @@ public void commitTransaction(CommitTransaction commitTx) {
                 .thenAccept(v -> connection.send(new TransactionCommitted(requestId, commitTx.getSegment(), commitTx.getTxid())));
 
         CompletableFuture.allOf(seal, merge)
-                .exceptionally(e -> handleException(requestId, transactionName, "Commit transaction", e));
+                .exceptionally(e -> {
+                    final Throwable cause;
+                    if (e instanceof CompletionException) {
+                        cause = e.getCause();
+                    } else {
+                        cause = e;
+                    }
+                    if (cause instanceof StreamSegmentMergedException) {
+                        log.warn("Stream segment is already merged '{}'.", transactionName);
```

Comment:
I think this should be INFO instead. WARN is for potentially problematic situations and here this is really harmless, unless we consider that multiple attempts to merge could be an indication of a problem.